src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java

```diff
@@ -358,40 +358,152 @@ private void doOptionField(String s) {
+        case "CompositeFormat":
+            return new CompositeFormat();
+        case "CreateBibORDFAuthors":
+            return new CreateBibORDFAuthors();
+        case "CreateDocBookAuthors":
+            return new CreateDocBookAuthors();
+        case "CurrentDate":
+            return new CurrentDate();
+        case "DOICheck":
+            return new DOICheck();
+        case "DOIStrip":
+            return new DOIStrip();
+        case "FirstPage":
+            return new FirstPage();
+        case "FormatPagesForHTML":
+            return new FormatPagesForHTML();
+        case "FormatPagesForXML":
+            return new FormatPagesForXML();
+        case "GetOpenOfficeType":
+            return new GetOpenOfficeType();
+        case "HTMLChars":
+            return new HTMLChars();
+        case "HTMLParagraphs":
+            return new HTMLParagraphs();
+        case "Iso690FormatDate":
+            return new Iso690FormatDate();
+        case "Iso690NamesAuthors":
+            return new Iso690NamesAuthors();
+        case "JournalAbbreviator":
+            return new JournalAbbreviator(repository);
+        case "LastPage":
+            return new LastPage();
+        case "FormatChars": // For backward compatibility
+        case "LatexToUnicode":
+            return new LatexToUnicodeFormatter();
+        case "NameFormatter":
+            return new NameFormatter();
+        case "NoSpaceBetweenAbbreviations":
+            return new NoSpaceBetweenAbbreviations();
+        case "Ordinal":
+            return new Ordinal();
+        case "RemoveBrackets":
+            return new RemoveBrackets();
+        case "RemoveBracketsAddComma":
+            return new RemoveBracketsAddComma();
+        case "RemoveLatexCommands":
+            return new RemoveLatexCommands();
+        case "RemoveTilde":
+            return new RemoveTilde();
+        case "RemoveWhitespace":
+            return new RemoveWhitespace();
+        case "RisKeywords":
+            return new RisKeywords();
+        case "RisMonth":
+            return new RisMonth();
+        case "RTFChars":
+            return new RTFChars();
+        case "ToLowerCase":
+            return new ToLowerCase();
+        case "ToUpperCase":
+            return new ToUpperCase();
+        case "XMLChars":
+            return new XMLChars();
+        case "Default":
+            return new Default();
+        case "FileLink":
+            return new FileLink();
+        case "Number":
+            return new Number();
+        case "RisAuthors":
+            return new RisAuthors();
+        case "Authors":
+            return new Authors();
+        case "IfPlural":
+            return new IfPlural();
+        case "Replace":
+            return new Replace();
+        case "WrapContent":
+            return new WrapContent();
+        case "WrapFileLinks":
+            return new WrapFileLinks();
+        default:
+            return new NotFoundFormatter(name);
```

Comment:
At first, I thought so, too. However a map would require that all formatters are initialized although only one of them is returned (and a static map is not possible since some of the formatters have constructor arguments).